recipes-extended/qemu-dm/qemu-dm/qemu-dm-wrapper

```diff
@@ -41,4 +41,25 @@ export QEMU_ALSA_VOL_CTRL=vm-${ALSA_VMID}
 
 ARGS="$*"
 
-exec /usr/bin/qemu-system-i386 $ARGS 2>&1 | logger -t qemu-$DOMID
+# We want to run `exec qemu | logger` with qemu replacing the current process
+# at the current PID.  That would actually run `exec qemu` in a subshell, so
+# We have to manually setup a pipe equilavent with a fifo and run the exec
+# directly from the main script.
+
+# Dummy commands to create a pipeline.
+sleep 9999 | sleep 9999 &
+# save the process ids
+PID2=$!
+PID1=$(jobs -p %+)
+# hijack the pipe's file descriptors using procfs
+exec 3</proc/$PID2/fd/0 4>/proc/$PID1/fd/1
```

Comment:
Per our side discussions and your testing, I would like to see the qemu-dm-wrapper here trap on the SIGHUP from libxl and exit the qemu and logger processes, instead of the anonymous pipe approach.

Reply:
Same here fwiw. Chris and I had talked about this offline as well.

Reply:
The trap/SIGHUP implementation is more straightforward, but this pipe stealing is kinda amazing...

Reply:
It is a pretty cool trick, honestly didn't know you could do that sort of things with pipes.

Reply:
trap/SIGHUP seems to be working, so I'm switching to that.